### ISSUE TITLE
ARM64 binaries for Windows, Linux and macOS platforms

### ISSUE THREAD

**Issue by zebrapurring** [Danny-Dasilva/CycleTLS#86]
*Requested feature* The NPM package currently ships with the prebuilt binaries for x86-64, would be great if you can add arm64 binaries as well. Or as a more generic solution, configure the `package.json` to build the binaries from source when installing the library, if prebuilt binaries are not available.

*Why the feature is needed* This would allow running the library on platforms such as a Raspberry Pi, or natively on Apple Silicon macs.

**Comment by White2001Offl** [comment]
You can do `npm run build:mac` which builds for your architecture. But you should have go installed

Same for `npm run build:windows` and `npm run build:linux`

**Comment by zebrapurring** [comment]
> You can do `npm run build:mac` which builds for your architecture. But you should have go installed
> 
> Same for `npm run build:windows` and `npm run build:linux`

Unfortunately this is not possible when using the package as a NPM dependency, since the sources are missing.

Also building the tool from source currently does not work because of #84.

**Comment by Danny-Dasilva** [comment]
I tested the default package on a raspberry pi 3 and 4 running the latest `Raspberry Pi OS` version and the package appeared to work just fine by default.

Anything that is a derivative of the three supported platforms should in theory function including a mac arm chip. 

For release  [0.0.19](https://github.com/Danny-Dasilva/CycleTLS/pull/87)  explicit support for linux `arm` and `arm64` was added.

You can see how it was implemented here

https://github.com/Danny-Dasilva/CycleTLS/blob/2a7fc20b8fe73f8f223433a7a7ce4e008951b70b/package.json#L15-L16

https://github.com/Danny-Dasilva/CycleTLS/blob/2a7fc20b8fe73f8f223433a7a7ce4e008951b70b/src/index.ts#L122-L128


If you are curious how to compile versions for a custom/niche operating system there is an added section in the README for it.

[**Cross Compiling for other platforms** ](https://github.com/Danny-Dasilva/CycleTLS#cross-compiling-for-other-platforms)

Basically you can compile using Golang's cross compilation tools and manually run the compiled file. The package should connect on the default port `9112` to the spawned Golang server, no need to update the `index.ts` file, you just need to run the compiled executable. 

e.g.
 ```bash
 $ ./index
 ```

**Comment by zebrapurring** [comment]
Great! Thanks for the quick response.

So the ARM binaries are not included in the NPM package, right? Do you think it would make sense to include them as well?

**Comment by Danny-Dasilva** [comment]
Correct, ARM binaries are not included in the NPM package, they should function on ARM without the specific binaries though (at least the arm chips I tested, Cortex-A53 and Cortex-A35). 

I can include them but then I need to build tests around them if they are supposed to be fully supported. Arm and Arm64 as well as other linux derivatives have complicated github actions implementations. If I can implement those I will include them in the npm package. Perhaps building a docker container would be preferred.
